lxd/instance/drivers/driver_lxc.go

```diff
@@ -6800,7 +6876,21 @@ func (d *lxc) FileSFTP() (*sftp.Client, error) {
 
 // stopForkFile attempts to send SIGINT to forkfile then waits for it to exit.
 func (d *lxc) stopForkfile() {
-	// Make sure that when the function exits, no forkfile is running by acquiring the lock (which indicates
+	// First, check if there is a SFTP process relying on this running forkfile
+	// and if so, stop it.
+	sftpMu.Lock()
+	p, found := sftpProcesses[d.forkfileRunningLockName()]
+	if found {
+		err := p.stop(context.Background())
+		if err != nil {
+			d.logger.Error("Trying to stop SFTP process but failed", logger.Ctx{"err": err})
+			return
+		}
+	}
+
+	sftpMu.Unlock()
+
+	// Then, make sure that when the function exits, no forkfile is running by acquiring the lock (which indicates
```

Comment:
The rest of this function should be moved into `p.stop()` as that should be responsible for stopping the process.

Reply:
I don't understand, `stopForkfile()` is called at a couple of places like `startCommon()` or `Snapshot()` and in these cases, the logic has nothing to do with an `sftpProcess` right ? So why, should we move the rest of the function in `p.stop()` ? And what if there is no `sftpProcess` running ?  `p.stop()` would never be called and therefore `stopForkfile()` would not delete or stop anything right ?